composer/trainer/trainer_hparams.py

```diff
@@ -421,10 +421,15 @@ def validate(self):
         super().validate()
 
         if self.deepspeed is not None:
-            self.deepspeed["zero_stage"] = cast(int, self.deepspeed.get("zero_stage", 0))
             self.deepspeed["steps_per_print"] = cast(int, self.deepspeed.get("steps_per_print", 1e20))
 
-            if self.deterministic_mode and self.deepspeed["zero_stage"] > 0:
+            zero_stage = 0
+            zero_config = cast(dict,
+                               self.deepspeed["zero_optimization"]) if "zero_optimization" in self.deepspeed else None
+            if zero_config:
+                zero_stage = cast(int, zero_config.get("stage"))
+
```

Comment:
The double ifs and use of zero_config is a bit awkward, what about:
```
if "zero_optimization" in self.deepspeed:
    zero_stage = cast(dict, self.deepspeed["zero_optimization"]).get("stage")
else:
    zero_stage = 0
```

Reply:
I like that, thanks for the suggestion.